sailfishusermanager.cpp

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserverd.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "sailfishusermanager.h"
+#include "usermanager_adaptor.h"
+
+#include <QDBusConnection>
+#include <QTimer>
+#include <QFile>
+#include <QDir>
+
+#include <pwd.h>
+#include <grp.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <libuser/user.h>
+
+const char* DBUS_INTERFACE = "org.sailfishos.usermanager";
+const char* DBUS_OBJECT_PATH = "/";
+const char* USER_GROUP = "users";
+const char* GROUPS_IDS_FILE = "/usr/lib/setup/group.ids";
+const char* GROUPS_SYSTEM = "SYSTEM_GROUPS";
+const char* GROUPS_USER = "USER_GROUPS";
+const char* SKEL_DIR = "/etc/skel";
```

Comment:
Most (or all?) of these should use QStringLiteral I think.

Reply:
Or, even if they would remain c-strings (e.g. to facilitate qFatal("%s") use), using "const char foo[] ..." or "const char * const foo ..." might be more appropriate.